.github/workflows/ci.yml

```diff
@@ -7,13 +7,13 @@ name: 💡🏠
 on: [pull_request]
 
 jobs:
-  ci:
-
+  # `basics` includes all non-smoke CI, except for unit and proto
+  basics:
     runs-on: ubuntu-latest
-    strategy:
-      # e.g. if lint fails, continue to the unit tests anyway
-      fail-fast: false
 
+    # A few steps are duplicated across all jobs. Can be done better when this feature lands:
+    #   https://github.community/t/reusing-sharing-inheriting-steps-between-jobs-declarations/16851
+    #   https://github.com/actions/runner/issues/438
     steps:
     - name: git clone
       uses: actions/checkout@v2
```

Comment:
faster to `with: fetch-depth: 100` at this point instead of splitting it with the `fetch` below?

Reply:
thx. i'll give that a shot.